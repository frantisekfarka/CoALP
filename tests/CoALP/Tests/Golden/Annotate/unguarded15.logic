q(s(s(Q1),Q2,Q3),1):-q(s(Q2,Q3,s(Q1)),2).
q(s(s(Q1),Q2,Q3),2):-q(s(Q2,Q3,s(Q1)),3).
q(s(s(Q1),Q2,Q3),3):-q(s(Q2,Q3,s(Q1)),1).

%?- q(s(s(X),s(X),s(X)),1).

% just checking cotrees/clauses:
%
% q(s(s(Q1),Q2,Q3),1)
% |
% q(s(Q2,Q3,s(Q1)),2)
% -------------
% guarded
%
% q(s(s(Q1),Q2,Q3),2)
% |
% q(s(Q2,Q3,s(Q1)),3)
% --------------
% guarded
%
% q(s(s(Q1),Q2,Q3),3)
% |
% q(s(Q2,Q3,s(Q1)),1)
% --------------
% guarded
%
% checking with unification
%
% q(s(s(Q1),Q2,Q3),1)
% |
% q(s(Q2,Q3,s(Q1)),2)
% 
% subst Q2/s(Q1'),
%
% q(s(s(Q1),s(Q1'),Q3),1)
% |
% q(s(s(Q1'),Q3,s(Q1)),2)
% |
% q(s(Q3,s(Q1),s(Q1')),3)
% -------------------------
% guarded because q(s(s(Q1),s(Q1'),Q3),1) :- q(s(s(Q1'),Q3,s(Q1)),2) is guarded.
%
% ...