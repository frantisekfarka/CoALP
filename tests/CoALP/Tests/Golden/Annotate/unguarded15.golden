q(s(s(V_2i), V_3i, V_4i), 1, transform-func-1(V_11i)) :- q(s(V_3i, V_4i, s(V_2i)), 2, V_11i).
q(s(s(V_5i), V_6i, V_7i), 2, transform-func-2(V_12c)) :- q(s(V_6i, V_7i, s(V_5i)), 3, V_12c).
q(s(s(V_8i), V_9i, V_10i), 3, transform-func-3(V_13c)) :- q(s(V_9i, V_10i, s(V_8i)), 1, V_13c).
